lib/did_you_mean/spell_checkers/method_name_checker.rb

```diff
@@ -59,6 +59,13 @@ def method_names
         method_names = receiver.methods + receiver.singleton_methods
         method_names += receiver.private_methods if @private_call
         method_names.uniq!
+        # Assume that people trying to use a writer are not interested in a reader
+        # and vice versa
+        if method_name =~ /=\Z/
+          method_names.select!{|name| name =~ /=\Z/}
+        else
+          method_names.reject!{|name| name =~ /=\Z/}
+        end
```

Comment:
```suggestion
        if method_name.match?(/=\Z/)
          method_names.select! { |name| name.match?(/=\Z/) }
        else
          method_names.reject! { |name| name.match?(/=\Z/) }
        end
```

This saves allocating useless MatchData objects. `did_you_mean` can be surprisingly perf sensitive on objects with lots of methods.

Reply:
Thanks!